src/declarative.js

```diff
@@ -107,7 +107,7 @@
     return moduleRecords[key] || (moduleRecords[key] = {
       key: key,
       dependencies: [],
-      module: new Module({}),
+      module: new Reflect.Module({}),
```

Comment:
> It may well be worth rewriting the build using ES modules and Rollup as another task here to make these scoping assumptions clearer!

just to make sure we're on the same wavelength, you're proposing rewriting src/*.js as ES6, and then use Rollup as build instead of concat to convert that into IIFE? I think that would be a very good idea - and would suggest extending to SystemJS as well.